### PR TITLE
libgit2: update enums from v1.8.0

### PR DESCRIPTION
**Pull request by inkydragon**

### CODE REVIEW

File: stdlib/LibGit2/src/error.jl [hunk 1]
```diff
@@ -27,14 +27,23 @@ export GitError
             EAPPLIED        = Cint(-18), # patch/merge has already been applied
             EPEEL           = Cint(-19), # the requested peel operation is not possible
             EEOF            = Cint(-20), # unexpected EOF
+            EINVALID        = Cint(-21), # Invalid operation or input
```
Comment (inkydragon) [inline stdlib/LibGit2/src/error.jl:30]: - `git_error_code`
    https://github.com/libgit2/libgit2/blob/v1.8.0/include/git2/errors.h#L50-L66

**Review by inkydragon — COMMENTED** [review]
- `/* Deprecated error values and functions */`
    https://github.com/libgit2/libgit2/blob/v1.8.0/src/util/errors.c#L376-L385

File: stdlib/LibGit2/src/error.jl [hunk 2]
```diff
@@ -27,14 +27,23 @@ export GitError
...
             EMISMATCH       = Cint(-33), # hashsum mismatch in object
             EINDEXDIRTY     = Cint(-34), # unsaved changes in the index would be overwritten
             EAPPLYFAIL      = Cint(-35), # patch application failed
             EOWNER          = Cint(-36), # the object is not owned by the current user
-            TIMEOUT         = Cint(-37)) # The operation timed out
+            TIMEOUT         = Cint(-37), # The operation timed out
+            EUNCHANGED      = Cint(-38), # There were no changes
+            ENOTSUPPORTED   = Cint(-39), # An option is not supported
+            EREADONLY       = Cint(-40), # The subject is read-only
+)
```
Comment (inkydragon) [inline stdlib/LibGit2/src/error.jl:46]:
```suggestion
)
```

**Comment by IanButterworth** [comment]
Does this fix this issue in https://github.com/JuliaLang/julia/pull/56786 https://buildkite.com/julialang/julia-master/builds/42893#0193accf-efd5-487b-9361-7c9152790fcd ?
```
  | 2024-12-09 14:47:52 EST | Pkg             (2) \|         failed at 2024-12-09T19:47:52.424
  | 2024-12-09 14:47:52 EST | Error During Test at /cache/build/tester-demeter6-10/julialang/julia-master/julia-e82f30c685/share/julia/stdlib/v1.12/Pkg/test/sources.jl:8
  | 2024-12-09 14:47:52 EST | Got exception outside of a @test
  | 2024-12-09 14:47:52 EST | ArgumentError: invalid value for Enum Code: -30848
  | 2024-12-09 14:47:52 EST | Stacktrace:
  | 2024-12-09 14:47:52 EST | [1] enum_argument_error(typename::Symbol, x::Int32)
  | 2024-12-09 14:47:52 EST | @ Base.Enums ./Enums.jl:93
  | 2024-12-09 14:47:52 EST | [2] Code
  | 2024-12-09 14:47:52 EST | @ ./Enums.jl:212 [inlined]
  | 2024-12-09 14:47:52 EST | [3] LibGit2.Error.GitError(err_code::Int32)
  | 2024-12-09 14:47:52 EST | @ LibGit2.Error /cache/build/tester-demeter6-10/julialang/julia-master/julia-e82f30c685/share/julia/stdlib/v1.12/LibGit2/src/error.jl:103
  | 2024-12-09 14:47:52 EST | [4] macro expansion
  | 2024-12-09 14:47:52 EST | @ /cache/build/tester-demeter6-10/julialang/julia-master/julia-e82f30c685/share/julia/stdlib/v1.12/LibGit2/src/error.jl:115 [inlined]
  | 2024-12-09 14:47:52 EST | [5] clone(repo_url::SubString{String}, repo_path::String, clone_opts::LibGit2.CloneOptions)
  | 2024-12-09 14:47:52 EST | @ LibGit2 /cache/build/tester-demeter6-10/julialang/julia-master/julia-e82f30c685/share/julia/stdlib/v1.12/LibGit2/src/repository.jl:470
```

**Comment by vtjnash** [comment]
That is an mbedtls error / error code

**Comment by inkydragon** [comment]
> Does this fix this issue in https://github.com/JuliaLang/julia/pull/56786 ?

No, That's another issue: #56763, maybe it's an upstream issue.

**Pull request merged**